opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchTextType.java

```diff
@@ -57,24 +56,18 @@ public boolean shouldCast(ExprType other) {
     return false;
   }
 
-  public Map<String, OpenSearchDataType> getFields() {
-    return fields;
-  }
-
   @Override
   protected OpenSearchDataType cloneEmpty() {
     return OpenSearchTextType.of(Map.copyOf(this.fields));
   }
 
-  /**
-   * Text field doesn't have doc value (exception thrown even when you call "get")
-   * Limitation: assume inner field name is always "keyword".
-   */
-  public static String convertTextToKeyword(String fieldName, ExprType fieldType) {
-    if (fieldType instanceof OpenSearchTextType
-        && ((OpenSearchTextType) fieldType).getFields().size() > 0) {
-      return fieldName + ".keyword";
+  @Override
+  public String convertFieldForSearchQuery(String fieldName) {
+    if (fields.size() == 0) {
+      return fieldName;
     }
-    return fieldName;
+    // Pick first field. What to do if there are multiple fields?
```

Comment:
Can you pass in the type like how it was with convertTextToKeyword and map that type by finding it in the list of fields?

Reply:
Is it needed?
Now this function isn't static and different types may overload it if needed. Having that we can avoid creating a new function like `convertXXXtoYYY` in future.

Reply:
Right. But how would it know which type to convert to? For example, doing aggregation on text with mapping
```
"textColumn": {
    "type": "text",
         "fields": {
            "date": {
               "type": "date"
             },
             "keyword": {
                "type": "keyword"
              }
          }
      }
}
```
will do aggregation on `textColumn.date`. What would be expected here?

Reply:
It is possible to aggregate on dates inside a text field.
I changed to find a string subfield if present in 8b0671c12.